### DISCUSSION TITLE
Ignore project subdirectory

### DISCUSSION THREAD

**Discussion by chriscraws** [zed-industries/zed#21250] [Feature Requests]
I have a git submodule inside my project directory. I get diagnostic results and search results from this directory. I'd love for zed to ignore it, but I don't know how.

**Comment by SomeoneToIgnore** [comment]
One way for that could be adding this directory to https://github.com/zed-industries/zed/blob/25f2cf9f6b0ed7d4a04773b8582c3c38e5e253d8/assets/settings/default.json#L666-L678 

If you copy the list into the local settings and add more entries, those entries won't be displayed anywhere in Zed's UI and should not get any diagnostics or appear in the search results.

**Reply by chriscraws** [reply]
Thank you! this definitely solves the issue of items appearing in the diagnostics and having tons of irrelevant entries in file search, however it does entirely hide the existence of the folder. The behavior of how Zed treats files that are in `.gitignore` is more ideal - where I can still browse and open files from zed, but they don't appear in search (unless they're open) and the folder/item is greyed out (but interactive) in the file explorer.

**Reply by SomeoneToIgnore** [reply]
It's a bit trickier with `.gitignore`-d files though, as such directories' contents is indexed when unfolded and that might bring in more things like the diagnostics.

**Reply by chriscraws** [reply]
I did a quick test by creating a Go file inside of a `.gitignore`'d directory and found editing worked normally (LSP features, formatting etc), the file became searchable after being opened (until Zed is restarted), and no issues appear inside project diagnostics whether the file has been opened or not. This would be pretty ideal for me to be able to configure for a submodule - but the solution you presented before works fine. It seems to me submodules you don't intend to edit are one of the only use-cases for something like this since you can't really `.gitignore` them.

**Reply by injust** [reply]
> It seems to me submodules you don't intend to edit are one of the only use-cases for something like this since you can't really `.gitignore` them.

@chriscraws A bit late, but if you have a folder `foo/` containing only submodules, you can gitignore `foo/*/*`. This will gitignore the contents of the submodules, but not the submodules themselves.

**Reply by chriscraws** [reply]
thanks, good tip!

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
